plugins/woocommerce/src/StoreApi/Utilities/AgenticCheckoutUtils.php

```diff
@@ -355,14 +353,51 @@ public static function is_authorized() {
 		return true;
 	}
 
+	/**
+	 * Validates a session.
+	 *
+	 * @param AgenticCheckoutSession $checkout_session Checkout session object.
+	 * @return void
+	 */
+	public static function validate( AgenticCheckoutSession $checkout_session ): void {
+		$messages = $checkout_session->get_messages();
+
+		// Check if ready for payment.
+		$needs_shipping = $checkout_session->get_cart()->needs_shipping();
+		$has_address    = WC()->customer && WC()->customer->get_shipping_address_1();
+
+		// Add info message if shipping is needed.
+		if ( $needs_shipping && ! $has_address ) {
+			$messages->add(
+				MessageError::missing(
```

Comment:
Better to use `info` for this, IMO. 
```suggestion
				MessageInfo::missing(
```

Reply:
Think of info messages as notices in Woo. You an add some info, but it is generic and forwarded to the user.

There is no such thing as `MessageInfo::missing()`, as info messages do not support error codes. They only have the message itself.